tests/static/js/formatters-internal/image.js

```diff
@@ -1,106 +1,181 @@
+      const newFileImgUrl = Formatters.image(newFileImg).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain,width=200');
+      const euFileImgUrl = Formatters.image(euFileImg).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain,width=200');
     });
 
-    it('when choosing the biggest image under threshold, use dimensions from the sourceUrl if width/height does not exist', () => {
-      const euImageUrl = Formatters.image(
-        { url: euUrl,
-          width: 1024,
-          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg'
-        }, 'x', false).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=516,height=384,fit=contain');
+    it('do not transform image regardless of atLeastAsLarge', () => {
+      let photoImgUrl = Formatters.image(photoImg, 'x', false).url;
+      expect(photoImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/fit=contain');
+      let photoWithPaddingImgUrl = Formatters.image(photoWithPaddingImg, '0x0', false).url;
+      expect(photoWithPaddingImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/1.0000/fit=contain');
+      let oldFileImgUrl = Formatters.image(oldFileImg, '0x', false).url;
+      expect(oldFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/0/FOO.jpg/fit=contain');
+      let newFileImgUrl = Formatters.image(newFileImg, 'x0', false).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain');
+      let euFileImgUrl = Formatters.image(euFileImg, 'x', false).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain');
+
+      photoImgUrl = Formatters.image(photoImg, 'x', true).url;
+      expect(photoImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/fit=contain');
+      photoWithPaddingImgUrl = Formatters.image(photoWithPaddingImg, 'x', true).url;
+      expect(photoWithPaddingImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/1.0000/fit=contain');
+      oldFileImgUrl = Formatters.image(oldFileImg, 'x', true).url;
+      expect(oldFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/0/FOO.jpg/fit=contain');
+      newFileImgUrl = Formatters.image(newFileImg, 'x', true).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain');
+      euFileImgUrl = Formatters.image(euFileImg, 'x', true).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain');
     });
+  });
 
-    it('when choosing the smallest image over threshold, omit width/height if can\'t parse it from the image object', () => {
-      const euImageUrl = Formatters.image({url: euUrl}, 'x', true).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=cover');
+  describe('when desiredSize is not parseable', () => {
+    it('throws an error', () => {
+      expect(() => {Formatters.image(photoImg, 'ax')}).toThrowError();
```

Comment:
can you also add a case for the height throwing an error (e.g. `100x-1`?

Reply:
done